contracts/contracts/strategies/MorphoCompoundStrategy.sol

```diff
@@ -0,0 +1,249 @@
+     */
+    function initialize(
+        address _vaultAddress,
+        address[] calldata _rewardTokenAddresses,
+        address[] calldata _assets,
+        address[] calldata _pTokens
+    ) external onlyGovernor initializer {
+        super._initialize(
+            MORPHO,
+            _vaultAddress,
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /**
+     * @dev Approve the spending of all assets by main Morpho contract,
+     *      if for some reason is it necessary.
+     */
+    function safeApproveAllTokens() external override {
+        uint256 assetCount = assetsMapped.length;
+        for (uint256 i = 0; i < assetCount; i++) {
+            address asset = assetsMapped[i];
+
+            // Safe approval
+            IERC20(asset).safeApprove(MORPHO, 0);
+            IERC20(asset).safeApprove(MORPHO, type(uint256).max);
+        }
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset
+     *      We need to approve and allow Morpho to move them
+     * @param _asset Address of the asset to approve
+     * @param _pToken The pToken for the approval
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(MORPHO, 0);
+        IERC20(_asset).safeApprove(MORPHO, type(uint256).max);
+    }
+
+    /**
+     * @dev Collect accumulated rewards and send them to Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        /**
+         * Gas considerations. We could query Morpho LENS's `getUserUnclaimedRewards` for each
+         * cToken separately and only claimRewards where it is economically feasible. Each call
+         * (out of 3) costs ~60k gas extra.
+         *
+         * Each extra cToken in the `poolTokens` of `claimRewards` function makes that call
+         * 89-120k more expensive gas wise.
+         *
+         * With Lens query in case where:
+         *  - there is only 1 reward token to collect. Net gas usage in best case would be
+         *    3*60 - 2*120 = -60k -> saving 60k gas
+         *  - there are 2 reward tokens to collect. Net gas usage in best case would be
+         *    3*60 - 120 = 60k -> more expensive for 60k gas
+         *  - there are 3 reward tokens to collect. Net gas usage in best case would be
+         *    3*60 = 180k -> more expensive for 180k gas
+         *
+         * For the above reasoning such "optimization" is not implemented
+         */
+
+        address[] memory poolTokens = new address[](assetsMapped.length);
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            poolTokens[i] = assetToPToken[assetsMapped[i]];
+        }
+
+        // slither-disable-next-line unused-return
+        IMorpho(MORPHO).claimRewards(
```

Comment:
`claimRewards` returns the amount claimed in case you need it

Reply:
thanks, yup I see that